lib/fetch/request.js

```diff
@@ -420,15 +420,9 @@ class Request {
       // 4. If headers is a Headers object, then for each header in its header
       // list, append header’s name/header’s value to this’s headers.
       if (headers instanceof Headers) {
-        // TODO (fix): Why doesn't this work?
-        // for (const [key, val] of headers[kHeadersList]) {
-        //   this[kHeaders].append(key, val)
-        // }
-
-        this[kState].headersList = new HeadersList([
-          ...this[kState].headersList,
-          ...headers[kHeadersList]
-        ])
+        for (const [key, val] of headers) {
```

Comment:
Also the spec says `for each header in its header list` so would maybe `of (headers[kHeadersList] || headers)`? Be an idea?

Reply:
And I think we can relax `headers instanceof Headers` to `headers.constructor.name === 'Headers'`

Reply:
I'm 95% sure that the Headers' iterator is the header list's iterator, although I'm not 100% sure with the new changes

Reply:
I can actually make the `headers instanceof Headers` a `headers !== init.headers` I'm pretty sure! I'll let the CI run before I commit the change.